### ISSUE TITLE
Cannot build with `plog_reader`

### ISSUE THREAD

**Issue by bridgeythegeek** [panda-re/panda#267]
Using the latest docker image.

As per [the instructions in plog_reader.cpp](https://github.com/panda-re/panda/blob/master/panda/src/plog_reader.cpp), I uncommented the two lines in [Makefile.panda.target](https://github.com/panda-re/panda/blob/master/panda/Makefile.panda.target) to enable the build:

```
# Uncomment next two lines to enable compiled plog_reader.
# if PLOG_READER is defined, plog writing functions in plog-cc.cpp will be disabled.
#PROGS+=$(PLOG_READER_PROG) 
#QEMU_CFLAGS += -DPLOG_READER
```

Config with:
```
./configure --target-list=i386-softmmu --extra-plugins-path=/my_plugins
```

And then `make` gave:
```
  GEN     config-host.h
  GEN     trace/generated-tcg-tracers.h
  GEN     trace/generated-helpers-wrappers.h
  GEN     trace/generated-helpers.h
  GEN     module_block.h
  CC      block.o
  GEN     i386-softmmu/config-target.h
  GEN     trace/generated-helpers.c
  LINK    i386-softmmu/qemu-system-i386
  CXX     i386-softmmu/panda/src/plog_reader.o
  LINK    i386-softmmu/plog_reader
panda/src/plog-cc.o: In function `rr_get_guest_instr_count':
/panda/panda/include/panda/rr/rr_log.h:138: undefined reference to `cpus'
panda/src/plog-cc.o: In function `PandaLog::write_entry(std::unique_ptr<panda::LogEntry, std::default_delete<panda::LogEntry> >)':
/panda/panda/src/plog-cc.cpp:303: undefined reference to `panda_in_main_loop'
/panda/panda/src/plog-cc.cpp:304: undefined reference to `cpus'
/panda/panda/src/plog-cc.cpp:304: undefined reference to `panda_current_pc'
collect2: error: ld returned 1 exit status
make[1]: *** [plog_reader] Error 1
make: *** [subdir-i386-softmmu] Error 2
```

**Comment by m000** [comment]
Do you really need to hardwire your plugin path?

I've found that setting EXTRA_PLUGINS_PATH environment variable to work
just fine.

The path pointed by the variable is expected to reflect the PANDA directory
structure. I.e. store your plugins under
$(EXTRA_PLUGINS_PATH)/panda/plugins. To enable compilation of a plugin, add
it to $(EXTRA_PLUGINS_PATH)/panda/plugins/config.panda.

I'm not sure how I came up with this solution. If it is not already
somewhere in the manual, it was probably some older mailing list message.
In that case, please do update the manual.


2018-02-27 21:32 GMT+01:00 Bridgey the Geek <notifications@github.com>:

**Comment by bridgeythegeek** [comment]
Hi,

Thanks for the reply.

I think there might be some confusion here. I'm not hard-wiring my plugin path. As per my `./configure` line, I am using `--extra-plugins-path`. Is that what you mean?

However, I'm not trying to compile a plugin. I'm trying to compile `plog_reader.cpp` - the application that would allow me to read the logs generated by `-pandalog <log-file>`.

Unless I've missed something..?

**Comment by wangray** [comment]
@bridgeythegeek I wrote the plog_reader and know why this is happening; let me explain and tell you the fix, and maybe you can suggest a better solution. 

The undefined references are all references from panda record/replay, which are included by the log-writing functions in the pandalog library (`plog-cc.cpp`). `plog_reader.cpp` imports the library but tells it to ignore the writing functions with `#ifndef PLOG_READER`. But for some reason, even after you set the `PLOG_READER` flag, `make` is still looking for (and unable to find) the record/replay symbols. I don't know how to get around this issue.

tl;dr If you make a new build directory with those two lines uncommented, plog_reader will compile. 

Alternatively, you could use the python plog_reader `panda/scripts/plog_reader.py`.

**Comment by moyix** [comment]
I believe this should be fixed now? Please re-open if I'm wrong!
